### PR TITLE
[flutter_driver] use mostly public screenshot API.

### LINKED ISSUE #130461: [Impeller] Add devicelab integration tests for all desktop backends.
Probably a benchmark running something non-trivial like gallery. We need this for macOS, windows, and Linux.

its unclear if the devicelab configuraiton will support non-software rendering.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Instead of completely private. This has been broken for Impeller for years, which shows how much this method is getting used.

Fixes https://github.com/flutter/flutter/issues/130461

### CODE REVIEW

**Comment by zanderso** [comment]
Nice. Is it possible to test this on CI?

File: dev/integration_tests/ui/test_driver/solid_color_test.dart [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    driver = await FlutterDriver.connect();
+    await driver.waitUntilFirstFrameRasterized();
+  });
+
+  test('Can render solid red', () async {
+    // RGBA Encoded Bytes.
+    final Uint8List data = (await driver.screenshot(format: ScreenshotFormat.rawStraightRgba)) as Uint8List;
+
+    expect(data[0] << 24 | data[1] << 16 | data[2] << 8 | data[3], 0xFF0000FF);
+  }, timeout: Timeout.none);
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply dev/integration_tests/ui/test_driver/solid_color_test.dart:23]: Its only needed to satisfy a lint check :)

File: packages/flutter_driver/test_driver/screenshot_test.dart [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    await driver.waitUntilFirstFrameRasterized();
+  });
+
+  test('it takes a screenshot', () async {
+    // PNG Encoded Bytes.
+    final Uint8List bytes = (await driver.screenshot()) as Uint8List;
+
+    // Check PNG header.
+    expect(bytes.sublist(0, 8), <int>[137, 80, 78, 71, 13, 10, 26, 10]);
+  });
```
[the remark this answers is not in the document]
Comment (jonahwilliams) [reply packages/flutter_driver/test_driver/screenshot_test.dart:25]: I'm currently investigating this, but it looks like none of these driver tests are running anywhere.

File: packages/flutter_driver/lib/src/driver/web_driver.dart [hunk 3]
```diff
@@ -181,7 +181,8 @@ class WebFlutterDriver extends FlutterDriver {
   }
 
   @override
-  Future<List<int>> screenshot() async {
+  Future<List<int>> screenshot({ScreenshotFormat format = ScreenshotFormat.png}) async {
+    assert(format == ScreenshotFormat.png, 'Web Driver only supports PNG screenshot format');
```
Comment (matanlurey) [inline packages/flutter_driver/lib/src/driver/web_driver.dart:185]: I would make this an `ArgumentError` (yeah all tests use assertions but driver can be used outside of tests).

**Review by matanlurey — APPROVED** [review]

File: packages/flutter_driver/lib/src/driver/web_driver.dart [hunk 4]
```diff
@@ -182,7 +182,9 @@ class WebFlutterDriver extends FlutterDriver {
 
   @override
   Future<List<int>> screenshot({ScreenshotFormat format = ScreenshotFormat.png}) async {
-    assert(format == ScreenshotFormat.png, 'Web Driver only supports PNG screenshot format');
+    if (format != ScreenshotFormat.png) {
+      throw ArgumentError('Web Driver only supports PNG screenshot format');
```
Comment (matanlurey) [inline packages/flutter_driver/lib/src/driver/web_driver.dart:186]: Use `ArgumentError.value` so the user sees what format was providwed.

**Review by matanlurey — APPROVED** [review]

File: packages/flutter_driver/lib/src/driver/web_driver.dart (hunk 4 above)
Comment (jonahwilliams) [reply packages/flutter_driver/lib/src/driver/web_driver.dart]: Cool

**Pull request merged**